framework/src/userobjects/ElementSubdomainModifier.C

```diff
@@ -224,40 +387,32 @@ ElementSubdomainModifier::updateBoundaryInfo(MooseMesh & mesh,
     for (auto side : elem->side_index_range())
     {
       const Elem * neighbor = elem->neighbor_ptr(side);
-      if (_complement_moving_boundary_specified)
-        bnd_info.remove_side(elem, side, _complement_moving_boundary_id);
+      bnd_info.remove_side(elem, side, _complement_moving_boundary_id);
       if (neighbor && neighbor != libMesh::remote_elem)
       {
         // If the neighbor has a different subdomain ID, then this side should be added to
         // the moving boundary
         if (neighbor->subdomain_id() != elem->subdomain_id())
         {
-          if (_moving_boundary_specified)
-            bnd_info.add_side(elem, side, _moving_boundary_id);
-          if (_complement_moving_boundary_specified)
+
+          unsigned int neighbor_side = neighbor->which_neighbor_am_i(elem);
+          bnd_info.add_side(neighbor, neighbor_side, _complement_moving_boundary_id);
+          if (neighbor->processor_id() != this->processor_id())
           {
-            unsigned int neighbor_side = neighbor->which_neighbor_am_i(elem);
-            bnd_info.add_side(neighbor, neighbor_side, _complement_moving_boundary_id);
-            if (neighbor->processor_id() != this->processor_id())
-            {
-              _ghost_sides_to_add[neighbor->processor_id()].emplace_back(neighbor->id(),
-                                                                         neighbor_side);
-            }
+            _ghost_sides_to_add[neighbor->processor_id()].emplace_back(neighbor->id(),
+                                                                       neighbor_side);
           }
         }
         // Otherwise remove this side and the neighbor side from the boundary.
         else
         {
-          if (_moving_boundary_specified || _complement_moving_boundary_specified)
-          {
-            // this will destroy any overlapping third side sets!
-            bnd_info.remove_side(elem, side);
-            unsigned int neighbor_side = neighbor->which_neighbor_am_i(elem);
-            bnd_info.remove_side(neighbor, neighbor_side);
-            if (neighbor->processor_id() != this->processor_id())
-              _ghost_sides_to_remove[neighbor->processor_id()].emplace_back(neighbor->id(),
-                                                                            neighbor_side);
-          }
+          // this will destroy any overlapping third side sets!
+          bnd_info.remove_side(elem, side, _complement_moving_boundary_id);
```

Comment:
this is a big improvement, as the previous code would indiscriminately delete all boundaries on the interface